src/Validator/CanonicalUrlValidator.php

```diff
@@ -0,0 +1,36 @@
+<?php
+declare(strict_types = 1);
+
+namespace Spaze\SecurityTxt\Validator;
+
+use Spaze\SecurityTxt\SecurityTxt;
+use Spaze\SecurityTxt\Violations\SecurityTxtCanonicalUrlMismatch;
+use Spaze\SecurityTxt\Violations\SecurityTxtSpecViolation;
+
+final class CanonicalUrlValidator
```

Comment:
Move the class uder the `Validator\Validators` namespace, like for example `ContactMissingFieldValidator`.

In the `SecurityTxtValidator` class add the instance to `$this->fieldValidators` then.

Then any modifications in the `SecurityTxtParser` should not be needed.